plugins/packages/restapi/lib/index.ts

```diff
@@ -169,9 +185,35 @@ export default class RestapiQueryService implements QueryService {
         ...paramsFromUrl,
         ...this.searchParams(sourceOptions, queryOptions, hasDataSource),
       },
-      ...(isUrlEncoded ? { form: json } : { json }),
     };
 
+    const hasFiles = Object.values(json || {}).some((item) => {
+      return isFileObject(item);
+    });
+
+    if (isUrlEncoded) {
+      requestOptions.form = json;
+    } else if (hasFiles) {
```

Comment:
can we add a check here for if the content type is multipart/form-data
because currently the Restapi plugin will not auto-add the content-type as multipart like browsers do